src/classes/i18n.js

```diff
@@ -199,15 +209,94 @@ export default class i18n {
     }
 
     get_lang_files (lang, cb) {
+        const MAX_RETRY = this.config.get('download_retry');
 
         // append a random string, to avoid getting a response from cache
         const random_str = '&rand=' + ~~(Math.random() * 1000);
         const url = this.translations_url.replace(':lang', lang) + random_str;
-        const write_stream = fs.createWriteStream(this.config.get('locale_dir') + lang + '.json');
 
-        request(url)
-            .pipe(write_stream)
-            .on('finish', cb);
+        const translation_file_path = this.config.get('locale_dir') + lang + '.json';
+
+        new Promise(async (resolve, reject) => {
+            if (
+                process.env.REFRESH_TRANSLATIONS
+                || !await this.version_match(translation_file_path)
+            ) {
+                // force re-download translations
+                await this.download_translations(url, translation_file_path);
+            }
+
+
+            let retry_count = 0;
+
+            async.doWhilst(
+                async () => {
+                    try {
+                        await this.check_file(translation_file_path);
+                        retry_count = MAX_RETRY;
+                    } catch (error) {
+                        await this.download_translations(url, translation_file_path);
+                        retry_count++;
+                    }
+
+                    return retry_count;
+                },
+                cb_res => cb_res < MAX_RETRY,
+                err => err? reject(err) : resolve()
+            );
+        })
+            .then(cb)
+            .catch(cb);
+    }
+
+    async open_translation_file (translation_file_path) {
+        /**
+         * Load file using readFile and JSON.parse. Don't use require as it
+         * caches loaded JSON
+         */
+        return JSON.parse(
```

Comment:
This would have already throw exception if JSON file is corrupted, so the test should be load the JSON and check for version, and empty

Reply:
Hmm, that won't really work properly in our case when the version do not match.i.e. when checking the version of the translation, once the json is called by `require`, its cached and then subsequent `require` would not loaded the new updated file.

Reply:
I'm not saying load with require... Use JSON parse and pass around the parsed object

Reply:
https://github.com/anyTV/anytv-i18n/pull/10/commits/a9889856b217c991ffaa87d0b148f482c28ab675